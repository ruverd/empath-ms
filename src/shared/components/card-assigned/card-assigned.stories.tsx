import type { Meta, StoryObj } from "@storybook/react";

import {
  MOCK_ASSIGNED_COURSES,
  MOCK_COURSES,
} from "@/shared/mocks/course.mock";
import { CardAssigned } from "./card-assigned";

const meta = {
  title: "Components/CardAssigned",
  component: CardAssigned,
  parameters: {
    layout: "centered",
  },
} satisfies Meta<typeof CardAssigned>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    ...MOCK_ASSIGNED_COURSES[0],
  },
};

export const WithoutHighlightTag: Story = {
  args: {
    course: {
      ...MOCK_COURSES[0],
      highlightTag: undefined,
    },
    percentageCompleted: 75,
    currentLesson: 5,
    isOnTrack: true,
    dueDate: new Date("2024-04-15"),
  },
};

export const WithoutDueDate: Story = {
  args: {
    course: MOCK_COURSES[0],
    percentageCompleted: 25,
    currentLesson: 2,
    isOnTrack: true,
  },
};

export const NotStarted: Story = {
  args: {
    course: {
      ...MOCK_COURSES[0],
      highlightTag: {
        title: "Required",
        variant: "primary" as const,
      },
    },
    percentageCompleted: 0,
    currentLesson: 1,
    isOnTrack: true,
    dueDate: new Date("2024-05-01"),
  },
};

export const AlmostComplete: Story = {
  args: {
    course: {
      ...MOCK_COURSES[0],
      highlightTag: {
        title: "Almost Done",
        variant: "primary" as const,
      },
    },
    percentageCompleted: 90,
    currentLesson: 9,
    isOnTrack: true,
    dueDate: new Date("2024-04-10"),
  },
};
