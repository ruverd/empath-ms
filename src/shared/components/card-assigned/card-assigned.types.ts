import type { CourseAssigned } from "@/shared/types/course.types";

export interface CardAssignedProps extends CourseAssigned {}