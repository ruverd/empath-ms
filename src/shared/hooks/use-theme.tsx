import { useContext } from "react";

import { ThemeProviderContext } from "@/shared/contexts/theme.context";

export const useTheme = () => {
  const context = useContext(ThemeProviderContext);

  if (context === undefined)
    throw new Error("useTheme must be used within a ThemeProvider");

  return context;
};
