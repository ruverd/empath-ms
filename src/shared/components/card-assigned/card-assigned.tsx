import { Status } from "@/shared/enums/status";
import { useTheme } from "@/shared/hooks/use-theme";
import { cn, formatDate } from "@/shared/lib/utils";
import { BadgeStatus } from "../badge-status";
import { ProgressCircle } from "../progress-circle";
import { Badge } from "../ui/badge";
import { Button } from "../ui/button";
import { Card, CardFooter, CardHeader, CardTitle } from "../ui/card";
import type { CardAssignedProps } from "./card-assigned.types";

export const CardAssigned = ({
  course,
  percentageCompleted = 0,
  currentLesson,
  isOnTrack,
  dueDate,
}: CardAssignedProps) => {
  const { theme } = useTheme();
  const isDark = theme === "dark";

  return (
    <Card
      className={cn(
        "border border-primary-violet bg-gradient-overlay-carrousel px-1.5 w-full py-2 flex flex-col rounded-2xl overflow-hidden",
        isDark && "border-primary-violet"
      )}
    >
      <CardHeader className="p-4 flex w-full flex-row items-start justify-between gap-2">
        <div className="flex items-start gap-2">
          <img
            src={course.imageUrl}
            alt="Course Image"
            className="size-[74px] aspect-square rounded-lg"
          />
          <CardTitle className="flex flex-col gap-1 p-0 !mt-0">
            {course?.highlightTag && (
              <div>
                <Badge
                  variant={
                    course.highlightTag.variant === "primary"
                      ? "default"
                      : "outline"
                  }
                  className="text-primary-cloud"
                >
                  {course.highlightTag.title}
                </Badge>
              </div>
            )}
            {currentLesson && !course?.highlightTag && (
              <span className="text-sm font-medium text-primary-cloud">{`Lesson ${currentLesson}`}</span>
            )}
            <h3 className="text-sm font-bold line-clamp-2 text-primary-cloud">
              {course.title}
            </h3>
            {dueDate && (
              <span className="text-xs text-primary-cloud">{`Due: ${formatDate(
                dueDate
              )}`}</span>
            )}
            {!!isOnTrack && <BadgeStatus variant={Status.ON_TRACK} />}
          </CardTitle>
        </div>
        <div className="flex items-start">
          <ProgressCircle percentage={percentageCompleted} />
        </div>
      </CardHeader>
      <CardFooter className="mt-auto py-2 px-2.5 flex items-center justify-end">
        <Button variant="outline" size="xs">
          Start Lesson
        </Button>
      </CardFooter>
    </Card>
  );
};
