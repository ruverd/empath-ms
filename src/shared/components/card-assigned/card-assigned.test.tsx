import * as useThemeHook from "@/shared/hooks/use-theme";
import {
  MOCK_ASSIGNED_COURSES,
  MOCK_COURSES,
} from "@/shared/mocks/course.mock";
import { render, screen } from "@testing-library/react";
import { beforeEach, describe, expect, it, vi } from "vitest";
import { CardAssigned } from "./card-assigned";

describe("CardAssigned", () => {
  beforeEach(() => {
    vi.spyOn(useThemeHook, "useTheme").mockReturnValue({
      theme: "light",
      setTheme: vi.fn(),
    });
  });

  it("should match light theme snapshot", () => {
    vi.spyOn(useThemeHook, "useTheme").mockReturnValue({
      theme: "light",
      setTheme: vi.fn(),
    });

    const { container } = render(
      <CardAssigned {...MOCK_ASSIGNED_COURSES[0]} />
    );
    expect(container).toMatchSnapshot();
  });

  it("should match dark theme snapshot", () => {
    vi.spyOn(useThemeHook, "useTheme").mockReturnValue({
      theme: "dark",
      setTheme: vi.fn(),
    });

    const { container } = render(
      <CardAssigned {...MOCK_ASSIGNED_COURSES[0]} />
    );
    expect(container).toMatchSnapshot();
  });

  it("should match snapshot without highlight tag", () => {
    const { container } = render(
      <CardAssigned
        course={{
          ...MOCK_COURSES[0],
          highlightTag: undefined,
        }}
        percentageCompleted={75}
        currentLesson={5}
        isOnTrack={true}
        dueDate={new Date("2024-04-15")}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("should match snapshot for not started course", () => {
    const { container } = render(
      <CardAssigned
        course={{
          ...MOCK_COURSES[0],
          highlightTag: {
            title: "Required",
            variant: "primary",
          },
        }}
        percentageCompleted={0}
        currentLesson={1}
        isOnTrack={true}
        dueDate={new Date("2024-05-01")}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("should match snapshot for almost complete course", () => {
    const { container } = render(
      <CardAssigned
        course={{
          ...MOCK_COURSES[0],
          highlightTag: {
            title: "Almost Done",
            variant: "primary",
          },
        }}
        percentageCompleted={90}
        currentLesson={9}
        isOnTrack={true}
        dueDate={new Date("2024-04-10")}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("should renders course title and image", () => {
    render(
      <CardAssigned
        course={MOCK_COURSES[0]}
        percentageCompleted={50}
        currentLesson={1}
        isOnTrack={true}
      />
    );

    expect(screen.getByText(MOCK_COURSES[0].title)).toBeInTheDocument();
    expect(screen.getByAltText("Course Image")).toHaveAttribute(
      "src",
      MOCK_COURSES[0].imageUrl
    );
  });

  it("should renders highlight tag when provided", () => {
    render(<CardAssigned {...MOCK_ASSIGNED_COURSES[0]} />);

    expect(
      screen.getByText(MOCK_ASSIGNED_COURSES[0].course?.highlightTag?.title!)
    ).toBeInTheDocument();
  });

  it("should renders current lesson when no highlight tag", () => {
    render(<CardAssigned {...MOCK_ASSIGNED_COURSES[1]} />);

    const lessonString = `Lesson ${MOCK_ASSIGNED_COURSES[1].currentLesson}`;

    expect(screen.getByText(lessonString)).toBeInTheDocument();
  });

  it("should renders due date when provided", () => {
    const dueDate = new Date("2024-04-30");
    render(<CardAssigned {...MOCK_ASSIGNED_COURSES[1]} />);

    expect(screen.getByText(/Due:/)).toBeInTheDocument();
  });

  it("should renders on track status when isOnTrack is true", () => {
    render(<CardAssigned {...MOCK_ASSIGNED_COURSES[0]} />);

    expect(screen.getByText("On Track")).toBeInTheDocument();
  });

  it("should renders progress circle with correct percentage", () => {
    render(<CardAssigned {...MOCK_ASSIGNED_COURSES[2]} />);

    const percentage = `${MOCK_ASSIGNED_COURSES[2].percentageCompleted}%`;
    expect(screen.getByText(percentage)).toBeInTheDocument();
  });

  it("should renders start lesson button", () => {
    render(<CardAssigned {...MOCK_ASSIGNED_COURSES[2]} />);

    expect(
      screen.getByRole("button", { name: "Start Lesson" })
    ).toBeInTheDocument();
  });
});
