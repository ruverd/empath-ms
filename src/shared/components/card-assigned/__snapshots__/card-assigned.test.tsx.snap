// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CardAssigned > should match dark theme snapshot 1`] = `
<div>
  <div
    class="text-card-foreground shadow-sm border bg-card-gradient px-1.5 w-full py-2 flex flex-col rounded-2xl overflow-hidden border-primary-violet"
  >
    <div
      class="space-y-1.5 p-4 flex w-full flex-row items-start justify-between gap-2"
    >
      <div
        class="flex items-start gap-2"
      >
        <img
          alt="Course Image"
          class="size-[74px] aspect-square rounded-lg"
          src="/src/assets/mock/course-1.png"
        />
        <div
          class="text-2xl font-semibold leading-none tracking-tight flex flex-col gap-1 p-0 !mt-0"
        >
          <div>
            <div
              class="inline-flex items-center border-shades-violet-5 rounded-full border px-2.5 py-1 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 bg-transparent text-primary-cloud"
            >
              Learning Pathway
            </div>
          </div>
          <h3
            class="text-sm font-bold line-clamp-2 text-primary-cloud"
          >
            CompTIA Security+ certified
          </h3>
          <div
            class="flex items-center gap-2"
          >
            <div
              class="size-2 rounded-full bg-secondary-green"
            />
            <span
              class="text-xs font-medium text-primary-cloud"
            >
              On Track
            </span>
          </div>
        </div>
      </div>
      <div
        class="flex items-start"
      >
        <div
          class="relative inline-flex items-center justify-center"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="rotate-[-90deg]"
            height="40"
            width="40"
          >
            <defs>
              <lineargradient
                id="progressGradient"
                x1="0%"
                x2="0%"
                y1="0%"
                y2="100%"
              >
                <stop
                  offset="0%"
                  stop-color="#5C15A7"
                />
                <stop
                  offset="100%"
                  stop-color="#0098BD"
                />
              </lineargradient>
            </defs>
            <circle
              cx="20"
              cy="20"
              fill="transparent"
              r="18"
              stroke="#2c004f"
              stroke-width="4"
            />
            <circle
              cx="20"
              cy="20"
              fill="transparent"
              r="18"
              stroke="url(#progressGradient)"
              stroke-dasharray="113.09733552923255 113.09733552923255"
              stroke-linecap="round"
              stroke-width="4"
              style="stroke-dashoffset: 108.57344210806325;"
            />
          </svg>
          <span
            class="absolute text-[#72e9ff] text-[10px] font-bold"
            style="transform: translate(-50%, -50%); left: 50%; top: 50%;"
          >
            4
            %
          </span>
        </div>
      </div>
    </div>
    <div
      class="p-6 mt-auto py-2 px-2.5 flex items-center justify-end"
    >
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-[40px] text-sm font-bold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-secondary-aqua text-secondary-aqua bg-transparent hover:bg-secondary-aqua hover:text-primary-indigo h-8 px-4"
      >
        Start Lesson
      </button>
    </div>
  </div>
</div>
`;

exports[`CardAssigned > should match light theme snapshot 1`] = `
<div>
  <div
    class="text-card-foreground shadow-sm border border-primary-violet bg-card-gradient px-1.5 w-full py-2 flex flex-col rounded-2xl overflow-hidden"
  >
    <div
      class="space-y-1.5 p-4 flex w-full flex-row items-start justify-between gap-2"
    >
      <div
        class="flex items-start gap-2"
      >
        <img
          alt="Course Image"
          class="size-[74px] aspect-square rounded-lg"
          src="/src/assets/mock/course-1.png"
        />
        <div
          class="text-2xl font-semibold leading-none tracking-tight flex flex-col gap-1 p-0 !mt-0"
        >
          <div>
            <div
              class="inline-flex items-center border-shades-violet-5 rounded-full border px-2.5 py-1 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 bg-transparent text-primary-cloud"
            >
              Learning Pathway
            </div>
          </div>
          <h3
            class="text-sm font-bold line-clamp-2 text-primary-cloud"
          >
            CompTIA Security+ certified
          </h3>
          <div
            class="flex items-center gap-2"
          >
            <div
              class="size-2 rounded-full bg-secondary-green"
            />
            <span
              class="text-xs font-medium text-primary-cloud"
            >
              On Track
            </span>
          </div>
        </div>
      </div>
      <div
        class="flex items-start"
      >
        <div
          class="relative inline-flex items-center justify-center"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="rotate-[-90deg]"
            height="40"
            width="40"
          >
            <defs>
              <lineargradient
                id="progressGradient"
                x1="0%"
                x2="0%"
                y1="0%"
                y2="100%"
              >
                <stop
                  offset="0%"
                  stop-color="#5C15A7"
                />
                <stop
                  offset="100%"
                  stop-color="#0098BD"
                />
              </lineargradient>
            </defs>
            <circle
              cx="20"
              cy="20"
              fill="transparent"
              r="18"
              stroke="#2c004f"
              stroke-width="4"
            />
            <circle
              cx="20"
              cy="20"
              fill="transparent"
              r="18"
              stroke="url(#progressGradient)"
              stroke-dasharray="113.09733552923255 113.09733552923255"
              stroke-linecap="round"
              stroke-width="4"
              style="stroke-dashoffset: 108.57344210806325;"
            />
          </svg>
          <span
            class="absolute text-[#72e9ff] text-[10px] font-bold"
            style="transform: translate(-50%, -50%); left: 50%; top: 50%;"
          >
            4
            %
          </span>
        </div>
      </div>
    </div>
    <div
      class="p-6 mt-auto py-2 px-2.5 flex items-center justify-end"
    >
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-[40px] text-sm font-bold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-secondary-aqua text-secondary-aqua bg-transparent hover:bg-secondary-aqua hover:text-primary-indigo h-8 px-4"
      >
        Start Lesson
      </button>
    </div>
  </div>
</div>
`;

exports[`CardAssigned > should match snapshot for almost complete course 1`] = `
<div>
  <div
    class="text-card-foreground shadow-sm border border-primary-violet bg-card-gradient px-1.5 w-full py-2 flex flex-col rounded-2xl overflow-hidden"
  >
    <div
      class="space-y-1.5 p-4 flex w-full flex-row items-start justify-between gap-2"
    >
      <div
        class="flex items-start gap-2"
      >
        <img
          alt="Course Image"
          class="size-[74px] aspect-square rounded-lg"
          src="/src/assets/mock/course-1.png"
        />
        <div
          class="text-2xl font-semibold leading-none tracking-tight flex flex-col gap-1 p-0 !mt-0"
        >
          <div>
            <div
              class="inline-flex items-center border-shades-violet-5 rounded-full border px-2.5 py-1 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 bg-shades-violet-3 text-primary-cloud"
            >
              Almost Done
            </div>
          </div>
          <h3
            class="text-sm font-bold line-clamp-2 text-primary-cloud"
          >
            CompTIA Security+ certified
          </h3>
          <span
            class="text-xs text-primary-cloud"
          >
            Due: Apr 9, 2024
          </span>
          <div
            class="flex items-center gap-2"
          >
            <div
              class="size-2 rounded-full bg-secondary-green"
            />
            <span
              class="text-xs font-medium text-primary-cloud"
            >
              On Track
            </span>
          </div>
        </div>
      </div>
      <div
        class="flex items-start"
      >
        <div
          class="relative inline-flex items-center justify-center"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="rotate-[-90deg]"
            height="40"
            width="40"
          >
            <defs>
              <lineargradient
                id="progressGradient"
                x1="0%"
                x2="0%"
                y1="0%"
                y2="100%"
              >
                <stop
                  offset="0%"
                  stop-color="#5C15A7"
                />
                <stop
                  offset="100%"
                  stop-color="#0098BD"
                />
              </lineargradient>
            </defs>
            <circle
              cx="20"
              cy="20"
              fill="transparent"
              r="18"
              stroke="#2c004f"
              stroke-width="4"
            />
            <circle
              cx="20"
              cy="20"
              fill="transparent"
              r="18"
              stroke="url(#progressGradient)"
              stroke-dasharray="113.09733552923255 113.09733552923255"
              stroke-linecap="round"
              stroke-width="4"
              style="stroke-dashoffset: 11.309733552923248;"
            />
          </svg>
          <span
            class="absolute text-[#72e9ff] text-[10px] font-bold"
            style="transform: translate(-50%, -50%); left: 50%; top: 50%;"
          >
            90
            %
          </span>
        </div>
      </div>
    </div>
    <div
      class="p-6 mt-auto py-2 px-2.5 flex items-center justify-end"
    >
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-[40px] text-sm font-bold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-secondary-aqua text-secondary-aqua bg-transparent hover:bg-secondary-aqua hover:text-primary-indigo h-8 px-4"
      >
        Start Lesson
      </button>
    </div>
  </div>
</div>
`;

exports[`CardAssigned > should match snapshot for not started course 1`] = `
<div>
  <div
    class="text-card-foreground shadow-sm border border-primary-violet bg-card-gradient px-1.5 w-full py-2 flex flex-col rounded-2xl overflow-hidden"
  >
    <div
      class="space-y-1.5 p-4 flex w-full flex-row items-start justify-between gap-2"
    >
      <div
        class="flex items-start gap-2"
      >
        <img
          alt="Course Image"
          class="size-[74px] aspect-square rounded-lg"
          src="/src/assets/mock/course-1.png"
        />
        <div
          class="text-2xl font-semibold leading-none tracking-tight flex flex-col gap-1 p-0 !mt-0"
        >
          <div>
            <div
              class="inline-flex items-center border-shades-violet-5 rounded-full border px-2.5 py-1 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 bg-shades-violet-3 text-primary-cloud"
            >
              Required
            </div>
          </div>
          <h3
            class="text-sm font-bold line-clamp-2 text-primary-cloud"
          >
            CompTIA Security+ certified
          </h3>
          <span
            class="text-xs text-primary-cloud"
          >
            Due: Apr 30, 2024
          </span>
          <div
            class="flex items-center gap-2"
          >
            <div
              class="size-2 rounded-full bg-secondary-green"
            />
            <span
              class="text-xs font-medium text-primary-cloud"
            >
              On Track
            </span>
          </div>
        </div>
      </div>
      <div
        class="flex items-start"
      >
        <div
          class="relative inline-flex items-center justify-center"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="rotate-[-90deg]"
            height="40"
            width="40"
          >
            <defs>
              <lineargradient
                id="progressGradient"
                x1="0%"
                x2="0%"
                y1="0%"
                y2="100%"
              >
                <stop
                  offset="0%"
                  stop-color="#5C15A7"
                />
                <stop
                  offset="100%"
                  stop-color="#0098BD"
                />
              </lineargradient>
            </defs>
            <circle
              cx="20"
              cy="20"
              fill="transparent"
              r="18"
              stroke="#2c004f"
              stroke-width="4"
            />
            <circle
              cx="20"
              cy="20"
              fill="transparent"
              r="18"
              stroke="url(#progressGradient)"
              stroke-dasharray="113.09733552923255 113.09733552923255"
              stroke-linecap="round"
              stroke-width="4"
              style="stroke-dashoffset: 113.09733552923255;"
            />
          </svg>
          <span
            class="absolute text-[#72e9ff] text-[10px] font-bold"
            style="transform: translate(-50%, -50%); left: 50%; top: 50%;"
          >
            0
            %
          </span>
        </div>
      </div>
    </div>
    <div
      class="p-6 mt-auto py-2 px-2.5 flex items-center justify-end"
    >
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-[40px] text-sm font-bold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-secondary-aqua text-secondary-aqua bg-transparent hover:bg-secondary-aqua hover:text-primary-indigo h-8 px-4"
      >
        Start Lesson
      </button>
    </div>
  </div>
</div>
`;

exports[`CardAssigned > should match snapshot without highlight tag 1`] = `
<div>
  <div
    class="text-card-foreground shadow-sm border border-primary-violet bg-card-gradient px-1.5 w-full py-2 flex flex-col rounded-2xl overflow-hidden"
  >
    <div
      class="space-y-1.5 p-4 flex w-full flex-row items-start justify-between gap-2"
    >
      <div
        class="flex items-start gap-2"
      >
        <img
          alt="Course Image"
          class="size-[74px] aspect-square rounded-lg"
          src="/src/assets/mock/course-1.png"
        />
        <div
          class="text-2xl font-semibold leading-none tracking-tight flex flex-col gap-1 p-0 !mt-0"
        >
          <span
            class="text-sm font-medium text-primary-cloud"
          >
            Lesson 5
          </span>
          <h3
            class="text-sm font-bold line-clamp-2 text-primary-cloud"
          >
            CompTIA Security+ certified
          </h3>
          <span
            class="text-xs text-primary-cloud"
          >
            Due: Apr 14, 2024
          </span>
          <div
            class="flex items-center gap-2"
          >
            <div
              class="size-2 rounded-full bg-secondary-green"
            />
            <span
              class="text-xs font-medium text-primary-cloud"
            >
              On Track
            </span>
          </div>
        </div>
      </div>
      <div
        class="flex items-start"
      >
        <div
          class="relative inline-flex items-center justify-center"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="rotate-[-90deg]"
            height="40"
            width="40"
          >
            <defs>
              <lineargradient
                id="progressGradient"
                x1="0%"
                x2="0%"
                y1="0%"
                y2="100%"
              >
                <stop
                  offset="0%"
                  stop-color="#5C15A7"
                />
                <stop
                  offset="100%"
                  stop-color="#0098BD"
                />
              </lineargradient>
            </defs>
            <circle
              cx="20"
              cy="20"
              fill="transparent"
              r="18"
              stroke="#2c004f"
              stroke-width="4"
            />
            <circle
              cx="20"
              cy="20"
              fill="transparent"
              r="18"
              stroke="url(#progressGradient)"
              stroke-dasharray="113.09733552923255 113.09733552923255"
              stroke-linecap="round"
              stroke-width="4"
              style="stroke-dashoffset: 28.27433388230814;"
            />
          </svg>
          <span
            class="absolute text-[#72e9ff] text-[10px] font-bold"
            style="transform: translate(-50%, -50%); left: 50%; top: 50%;"
          >
            75
            %
          </span>
        </div>
      </div>
    </div>
    <div
      class="p-6 mt-auto py-2 px-2.5 flex items-center justify-end"
    >
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-[40px] text-sm font-bold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-secondary-aqua text-secondary-aqua bg-transparent hover:bg-secondary-aqua hover:text-primary-indigo h-8 px-4"
      >
        Start Lesson
      </button>
    </div>
  </div>
</div>
`;
